Microsoft.Toolkit.Uwp.UI/ImageCache/ImageCache.cs

```diff
@@ -188,16 +245,18 @@ public static string GetCacheFileName(Uri uri)
             return image;
         }
 
-        private static async Task<BitmapImage> GetFromCacheOrDownloadAsync(Uri uri)
+        private static async Task<BitmapImage> GetFromCacheOrDownloadAsync(Uri uri, string key, bool isPreCache)
         {
             BitmapImage image = null;
             DateTime expirationDate = DateTime.Now.Subtract(CacheDuration);
 
-            var key = GetCacheFileName(uri);
-
             if (MaxMemoryCacheSize > 0)
             {
                 image = GetFromMemoryCache(key);
+                if (isPreCache && image != null)
+                {
+                    return null;
```

Comment:
IMO remove the if statement and return the image.. if the caller is expecting image then great otherwise it will be ignored